cmake/list_insert_options.cmake

```diff
@@ -0,0 +1,55 @@
+#
+# Append key[=value] elements from one list to another list if they are not already in the target list.
```

Comment:
>  You already have the same problem with two different values both coming from dependencies because you have little control in which order they are being found.

[Yo have proposed](https://github.com/ros/catkin/issues/610#issuecomment-270165325) that in this case the dependencies must be resolved separately and cannot use the combined compile options.

> I think it is just counter intuitive if this function uses the first value.

For me it's not counter intuitive, it resembles the C++ `map::insert` logic